### PR TITLE
WIP adding symbols to _.omit

### PR DESCRIPTION
**Pull request by edwardkerry**
NOT COMPLETED

This is a work in progress, attempting to address
https://github.com/jashkenas/underscore/issues/2457

Any guidance on how best to approach this would be welcome. I have not been successful in passing the objects symbols through the function.

### CODE REVIEW

File: test/objects.js [hunk 1]
```diff
@@ -243,6 +243,15 @@
     assert.deepEqual(_.omit(data, function(val, key) {
       return this[key] === 3 && this === instance;
     }, instance), {a: 1, b: 2}, 'function is given context');
+
+    var Foo = function(){
+      this.a = 0;
+      this[symbol1] = 1;
+    }
+    var symbol1 = Symbol('sym1');
+    var foo = new Foo();
```
Comment (jdalton) [inline test/objects.js]: Be sure to add
```js
var symbol2 = Symbol('sym2');
Foo.prototype[symbol2] = 2;
```
before
```js
var var foo = new Foo();
```

File: underscore.js [hunk 2]
```diff
@@ -1100,6 +1100,9 @@
 
    // Return a copy of the object without the blacklisted properties.
   _.omit = restArgs(function(obj, keys) {
+    if(obj != null && Object.getOwnPropertySymbols(obj) != null){
```
Comment (akre54) [inline underscore.js]: `Object.getOwnPropertySymbols` is a relatively recent addition. IE doesn't support it yet, for instance. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertySymbols#Browser_compatibility

Comment (jdalton) [inline underscore.js]: IE won't as there's no more IE after 11. Edge 12 and 13 do though.

Comment (akre54) [inline underscore.js]: Regardless, FF and Chrome both added support only about a year ago, which is typically less than our support window. I think it's less of an issue in evergreen browsers (Edge is evergreen, right?) but probably something we should figure out in another thread.

Regardless, @edwardkerry maybe add a check that `Object.ownPropertySymbols` exists before using it?

Comment (jdalton) [inline underscore.js]: > Regardless, FF and Chrome both added support only about a year ago, which is typically less than our support window.

The browsers with symbol support have the method so there's not an issue of inconsistent support. If they support symbols then it'll work. Yes they'll need to be method existence checks.

Comment (jdalton) [inline underscore.js]: > (Edge is evergreen, right?) 

Yep!

**Pull request closed without merging**